example/data-persistence-tf/main.tf

```diff
@@ -51,36 +8,31 @@ provider "aws" {
   region = var.aws_region
 }
 
-module "data_persistence" {
-  source = "../../tf-modules/data-persistence"
+resource "random_string" "db_pass" {
+  length  = 20
+  upper   = true
+  special = false
+}
 
-  prefix                     = var.prefix
-  subnet_ids                 = var.subnet_ids
+module "provision_database" {
```

Comment:
So this Lambda invocation modle lives in data-persistence, because even though the cluster is created by a separate top level module to facilitate sharing a cluster across deployments, the database itself is unique to a deployment?

Reply:
Correct.    

This is analogous to how dynamo works - except AWS provisions Dynamo, and we create tables.   In postgres, they're just organized into "databases" within the database instance, and in theory also cross-organized by schema/schema perms, etc.    It made sense (to me) to handle 'database setup' here.  

The one thing this *does not* handle well is database deletion - but given that's a nearly trivial DBA task and this isn't the user module I chose not to address that here.